src/language-js/comments.js

```diff
@@ -911,6 +918,27 @@ function handleTSMappedTypeComments(
   return false;
 }
 
+function handleEndOfLineBlockComments(
+  text,
+  enclosingNode,
+  followingNode,
+  comment,
+  options
+) {
+  if (
+    enclosingNode &&
+    (enclosingNode.type === "LogicalExpression" ||
+      enclosingNode.type === "BinaryExpression") &&
+    followingNode &&
+    isBlockComment(comment) &&
+    privateUtil.hasNewline(text, options.locEnd(comment))
```

Comment:
Why is this condition needed? This function is called from `handleEndOfLineComment`. Isn't the comment by this point already classified as end-of-line?

Reply:
That's exactly. This is fixed on https://github.com/prettier/prettier/pull/7787/commits/a6dabaf7d0a05b023225be76307104f2a361d282.